### PR TITLE
fix bug in lock test

### PR DESCRIPTION
**Pull request by themanforfree**
Please briefly answer these questions:

* what problem are you trying to solve? (or if there's no problem, what's the motivation for this change?)
fix bug in lock test
* what changes does this pull request make?
generate unique lease id instead use timestamp
* are there any non-obvious implications of these changes? (does it break compatibility with previous versions, etc)
no

### CODE REVIEW

File: xline/src/id_gen.rs [hunk 1]
```diff
@@ -38,6 +38,11 @@ impl IdGenerator {
         let suffix = self.suffix.fetch_add(1, Ordering::Release);
```
Comment (Phoenix500526) [inline xline/src/id_gen.rs:38]: `Relaxed` is enough.

File: xline/src/id_gen.rs [hunk 2]
```diff
@@ -38,6 +38,11 @@ impl IdGenerator {
...
         self.prefix | suffix
     }
+
+    /// Generate next id and cast to i64
+    pub(crate) fn next_as_i64(&self) -> i64 {
```
Comment (Phoenix500526) [inline xline/src/id_gen.rs:43]: Why don't you inline the `next` method here and rename the `next_as_i64` as `next`? Doing so can increase coverage.

**Review by Phoenix500526 — CHANGES_REQUESTED** [review]

File: xline/src/id_gen.rs (hunk 1 above)
Comment (themanforfree) [reply xline/src/id_gen.rs]: Fixed

File: xline/src/id_gen.rs (hunk 2 above)
Comment (themanforfree) [reply xline/src/id_gen.rs]: Fixed

**Review by Phoenix500526 — APPROVED** [review]

File: xline/src/id_gen.rs [hunk 3]
```diff
@@ -34,9 +34,10 @@ impl IdGenerator {
     }
 
     /// Generate next id
-    pub(crate) fn next(&self) -> u64 {
-        let suffix = self.suffix.fetch_add(1, Ordering::Release);
-        self.prefix | suffix
+    pub(crate) fn next(&self) -> i64 {
```
Comment (rogercloud) [inline xline/src/id_gen.rs:37]: A test to cover this change?

File: xline/src/server/lock_server.rs [hunk 4]
```diff
@@ -252,11 +236,7 @@ where
 
     /// Lease grant
     async fn lease_grant(&self, token: Option<String>) -> Result<i64, tonic::Status> {
-        let lease_id = SystemTime::now()
-            .duration_since(UNIX_EPOCH)
-            .unwrap_or_else(|e| panic!("SystemTime before UNIX EPOCH! {e}"))
-            .as_secs()
-            .cast(); // TODO: generate lease unique id
+        let lease_id = self.id_gen.next();
```
Comment (rogercloud) [inline xline/src/server/lock_server.rs:239]: Have we double checked other id generations?

File: xline/src/id_gen.rs (hunk 3 above)
Comment (themanforfree) [reply xline/src/id_gen.rs:37]: already cover in the back

**Review by markcty — APPROVED** [review]

**Pull request merged**